scripts/generate_rust_analyzer.py

```diff
@@ -65,7 +65,7 @@ def append_crate(display_name, root_module, deps, cfg=[], is_workspace_member=Tr
         [],
         is_proc_macro=True,
     )
-    crates[-1]["proc_macro_dylib_path"] = "rust/libmacros.so"
+    crates[-1]["proc_macro_dylib_path"] = str((objtree / "rust/libmacros.so").resolve(True))
```

Comment:
Is `.resolve()` necessary?

Reply:
Yes, during my testing, `objtree` sometimes is empty, the reason is unknown.

Reply:
In which case is it empty where you need absolute paths?

Reply:
as shown here
<img width="1404" alt="image" src="https://user-images.githubusercontent.com/41831480/196454362-82c09dbb-d871-420d-87a0-d3661a8af5a6.png">

Reply:
> In which case is it empty where you need absolute paths?

I copy the generated `rust-project.json` from build tree to my out-of-tree module directory, the complaint will show up.

Reply:
You did have to run this script for your out of tree module rather than copying the in-tree rust-project.json file anyway to handle the out of tree module itself, right?

Reply:
In the theory yes, but currently the kernel Makefile does not support calling it in module directory with `make KPATH=<path-to-build-root> M=<path-to-module>`

Maybe I will add that support later

Reply:
Right, https://github.com/Rust-for-Linux/linux/pull/914 would add support for this AFAICT. Looks like it also fixes proc macros.